libguard/guard_interface.cpp

```diff
@@ -178,17 +193,32 @@ GuardRecord create(const EntityPath& entityPath, uint32_t eId, uint8_t eType)
 
     // Space left in GUARD file before writing a new record
     avalSize = file.size() - ((lastPos + 1) * sizeof(existGuard) + headerSize);
-	if (avalSize < sizeof(existGuard))
+    if (avalSize < sizeof(existGuard))
     {
-        guard_log(GUARD_ERROR, "Size left in GUARD file is %d\n", avalSize);
-        throw std::runtime_error("Enough size is not available in GUARD file");
+        if ((empPos == 0) && !isResolvedRecordPresent)
+        {
+            guard_log(GUARD_ERROR, "Size left in GUARD file is %d\n", avalSize);
+            throw std::runtime_error(
+                "Enough size is not available in GUARD file");
+        }
+
+        resolveRecordClear = true;
     }
 
     GuardRecord guard;
     //! if blank record exist
     if (isBlankRecord(existGuard))
```

Comment:
When this case will occur? Just looking for the code which you added in this block for this patch.

Reply:
This case will not happen, as the size check we are doing above.

Reply:
JS: Do you want to remove it then?